engineio/src/socket.rs

```diff
@@ -103,17 +113,37 @@ impl Socket {
     }
 
     /// Polls for next payload
-    pub(crate) fn poll(&self) -> Result<Option<Payload>> {
-        if self.connected.load(Ordering::Acquire) {
-            let data = self.transport.as_transport().poll()?;
-
-            if data.is_empty() {
+    pub(crate) fn poll(&self) -> Result<Option<Packet>> {
+        loop {
+            if self.connected.load(Ordering::Acquire) {
+                if self.remaining_packets.read()?.is_some() {
+                    // SAFETY: checked is some above
+                    let mut iter = self.remaining_packets.write()?;
+                    let iter = iter.as_mut().unwrap();
+                    if let Some(packet) = iter.next() {
+                        return Ok(Some(packet));
+                    } else {
+                        // Iterator has run out of packets, get a new payload
```

Comment:
Is that a TODO?

Reply:
ad5513e5ae6ce426e6851da23c9336664e32d2c0